crates/simulacrum/src/lib.rs

```diff
@@ -126,11 +132,20 @@ impl<R> Simulacrum<R> {
     /// If the above checks are successful then the transaction is immediately executed, enqueued
     /// to be included in the next checkpoint (the next time `create_checkpoint` is called) and the
     /// corresponding TransactionEffects are returned.
-    pub fn execute_transaction(&mut self, transaction: Transaction) -> Result<TransactionEffects> {
-        // This only supports traditional authenticators and not zklogin
+    pub fn execute_transaction(
+        &mut self,
+        transaction: Transaction,
+    ) -> anyhow::Result<TransactionEffects> {
+        Ok(self.execute_txn_impl(transaction)?.0)
+    }
+
+    pub fn execute_txn_impl(
```

Comment:
What are these execution errors used for? Are these just more details around how/why a transaction failed? We could probably do with a slightly better name here to indicate what it's doing and how it's different from the other function.

Reply:
> Are these just more details around how/why a transaction failed? 

Yes, thats quite important to know. The original implementation disregards the execution error.
My preference would actually be to have only one function which returns both effects and error. Doesn't make sense to have both. So I'll modify the original to return exec error.